lib/editor-settings.php

```diff
@@ -28,6 +28,9 @@ function gutenberg_extend_post_editor_settings( $settings ) {
 		$settings['defaultTemplatePartAreas'] = gutenberg_get_allowed_template_part_areas();
 	}
 
+	// Some blocks have to use absolute url.
+	$settings['adminUrl'] = is_callable( 'admin_url' ) ? admin_url() : '';
```

Comment:
`function_exists` is the default pattern for determining if a function is in memory for use

I'm wondering: Should a REST endpoint be used instead of the backend admin URL?

Gutenberg is using the REST API and is not loading the full backend functionality. I'm wondering if using `admin_url()` here, assuming it's loaded into memory at this point, might expose unintentional leaks. 

Hmm not sure. Raising the point for consideration. cc @azaozz @TimothyBJacobs

Reply:
I don't mind using `function_exists`.
Fixed in 198191eccffc2d94da2f214bf991e2b33ccab900

Reply:
There is a ticket about creating a REST API endpoint for this whole thing, I'm not sure what the expected timing on that is, https://github.com/WordPress/wordpress-develop/pull/1508. CC: @adamziel.

Reply:
Should we wait for this simple fix to land after we have that endpoint?

This fix will be ported into core, 100% that function will be there. It is not a feature of the plugin, it is a feature of how core initializes the editors. We've used this in e.g. `modify_welcome_panel` to modify links in WP Admin with no issues.

So I think we can safely use it as it does not load more resources, it does not tie Gutenberg to core more, and it's a quick fix for other problems of the future: having the admin url available as a setting,

Reply:
I was away for last two weeks but I'm back now. I will fix the test failures on https://github.com/WordPress/wordpress-develop/pull/1508 and then we'll only be an approval away from merging.

Reply:
@draganescu
> Should we wait for this simple fix to land after we have that endpoint?

Maybe I'm not getting it, but I'd not merge this ticket because, as @hellofromtonya mentioned, it will expose the backend admin URL. Instead, I'd wait until @adamziel's PR is merged. Then we can implement the fix using a REST endpoint.

Reply:
I fixed the tests as promised, all the reviews are welcome :D